from PyQt5 import QtCore, QtGui, QtWidgets
from Backend.HookCollectionManager import HookCollectionManager
from Backend.HookCollection import HookCollection

class Ui_hookCollectionDialog(object):

    def __init__(self, hookCollectionManager, hookCollectionView):
        self.hookCollectionManager = hookCollectionManager
        #self.hookCollectionView = self.hookCollectionView

    def setupUi(self, hookCollectionDialog):
        hookCollectionDialog.setObjectName("hookCollectionDialog")
        hookCollectionDialog.resize(400, 300)
        self.verticalLayout_2 = QtWidgets.QVBoxLayout(hookCollectionDialog)
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")
        self.formEditFrame = QtWidgets.QFrame(hookCollectionDialog)
        self.formEditFrame.setFrameShape(QtWidgets.QFrame.Panel)
        self.formEditFrame.setFrameShadow(QtWidgets.QFrame.Plain)
        self.formEditFrame.setObjectName("formEditFrame")
        self.verticalLayout_3 = QtWidgets.QVBoxLayout(self.formEditFrame)
        self.verticalLayout_3.setObjectName("verticalLayout_3")
        self.formLayout_3 = QtWidgets.QFormLayout()
        self.formLayout_3.setObjectName("formLayout_3")
        self.label = QtWidgets.QLabel(self.formEditFrame)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.formLayout_3.setWidget(0, QtWidgets.QFormLayout.LabelRole, self.label)
        self.hookCollectionNameEdit = QtWidgets.QLineEdit(self.formEditFrame)
        self.hookCollectionNameEdit.setObjectName("hookCollectionNameEdit")
        self.formLayout_3.setWidget(0, QtWidgets.QFormLayout.FieldRole, self.hookCollectionNameEdit)
        self.label_2 = QtWidgets.QLabel(self.formEditFrame)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.formLayout_3.setWidget(1, QtWidgets.QFormLayout.LabelRole, self.label_2)
        self.descriptionEdit = QtWidgets.QLineEdit(self.formEditFrame)
        self.descriptionEdit.setObjectName("descriptionEdit")
        self.formLayout_3.setWidget(1, QtWidgets.QFormLayout.FieldRole, self.descriptionEdit)
        self.label_3 = QtWidgets.QLabel(self.formEditFrame)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.formLayout_3.setWidget(2, QtWidgets.QFormLayout.LabelRole, self.label_3)
        self.statusComboBox = QtWidgets.QComboBox(self.formEditFrame)
        self.statusComboBox.setObjectName("statusComboBox")
        self.formLayout_3.setWidget(2, QtWidgets.QFormLayout.FieldRole, self.statusComboBox)
        self.label_4 = QtWidgets.QLabel(self.formEditFrame)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.formLayout_3.setWidget(3, QtWidgets.QFormLayout.LabelRole, self.label_4)
        self.executionSequenceEdit = QtWidgets.QLineEdit(self.formEditFrame)
        self.executionSequenceEdit.setObjectName("executionSequenceEdit")
        self.formLayout_3.setWidget(3, QtWidgets.QFormLayout.FieldRole, self.executionSequenceEdit)
        self.verticalLayout_3.addLayout(self.formLayout_3)
        self.verticalLayout.addWidget(self.formEditFrame)
        self.verticalLayout_2.addLayout(self.verticalLayout)
        self.hookEditFrame = QtWidgets.QFrame(hookCollectionDialog)
        self.hookEditFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.hookEditFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.hookEditFrame.setObjectName("hookEditFrame")
        self.verticalLayout_4 = QtWidgets.QVBoxLayout(self.hookEditFrame)
        self.verticalLayout_4.setObjectName("verticalLayout_4")
        self.scrollArea = QtWidgets.QScrollArea(self.hookEditFrame)
        self.scrollArea.setWidgetResizable(True)
        self.scrollArea.setObjectName("scrollArea")
        self.scrollAreaWidgetContents = QtWidgets.QWidget()
        self.scrollAreaWidgetContents.setGeometry(QtCore.QRect(0, 0, 372, 126))
        self.scrollAreaWidgetContents.setObjectName("scrollAreaWidgetContents")
        self.gridLayout_2 = QtWidgets.QGridLayout(self.scrollAreaWidgetContents)
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.verticalScrollBar = QtWidgets.QScrollBar(self.scrollAreaWidgetContents)
        self.verticalScrollBar.setOrientation(QtCore.Qt.Vertical)
        self.verticalScrollBar.setObjectName("verticalScrollBar")
        self.gridLayout_2.addWidget(self.verticalScrollBar, 0, 1, 1, 1)
        self.gridLayout = QtWidgets.QGridLayout()
        self.gridLayout.setObjectName("gridLayout")
        self.statusListView = QtWidgets.QListView(self.scrollAreaWidgetContents)
        self.statusListView.setObjectName("statusListView")
        self.gridLayout.addWidget(self.statusListView, 1, 1, 1, 1)
        self.hookExecutionListView = QtWidgets.QListView(self.scrollAreaWidgetContents)
        self.hookExecutionListView.setObjectName("hookExecutionListView")
        self.gridLayout.addWidget(self.hookExecutionListView, 1, 2, 1, 1)
        self.label_7 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_7.setFont(font)
        self.label_7.setObjectName("label_7")
        self.gridLayout.addWidget(self.label_7, 0, 2, 1, 1)
        self.hookListView = QtWidgets.QListView(self.scrollAreaWidgetContents)
        self.hookListView.setObjectName("hookListView")
        self.gridLayout.addWidget(self.hookListView, 1, 0, 1, 1)
        self.label_6 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.gridLayout.addWidget(self.label_6, 0, 1, 1, 1)
        self.label_5 = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.gridLayout.addWidget(self.label_5, 0, 0, 1, 1)
        self.gridLayout_2.addLayout(self.gridLayout, 0, 0, 1, 1)
        self.scrollArea.setWidget(self.scrollAreaWidgetContents)
        self.verticalLayout_4.addWidget(self.scrollArea)
        self.verticalLayout_2.addWidget(self.hookEditFrame)
        self.widget = QtWidgets.QWidget(hookCollectionDialog)
        self.widget.setObjectName("widget")
        self.horizontalLayout_5 = QtWidgets.QHBoxLayout(self.widget)
        self.horizontalLayout_5.setObjectName("horizontalLayout_5")
        spacerItem = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.horizontalLayout_5.addItem(spacerItem)
        self.saveButton = QtWidgets.QPushButton(self.widget)
        self.saveButton.setObjectName("saveButton")
        self.horizontalLayout_5.addWidget(self.saveButton)
        self.horizontalLayout_4 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_4.setObjectName("horizontalLayout_4")
        self.cancelButton = QtWidgets.QPushButton(self.widget)
        self.cancelButton.setObjectName("cancelButton")
        self.horizontalLayout_4.addWidget(self.cancelButton)
        self.horizontalLayout_5.addLayout(self.horizontalLayout_4)
        self.verticalLayout_2.addWidget(self.widget)
        self.statusComboBox.addItem('Enabled')
        self.statusComboBox.addItem('Disabled')
        self.retranslateUi(hookCollectionDialog)
        QtCore.QMetaObject.connectSlotsByName(hookCollectionDialog)

    def setupSignals(self):
        pass
        #self.saveButton.clicked(self.saveButtonClicked)

    def saveButtonClicked(self):
        description = self.descriptionEdit.text()
        hookCollectionName = self.hookCollectionNameEdit.text()
        status = self.statusComboBox.currentText()
        executionSequence = self.executionSequenceEdit.text()
        self.hookCollectionView.hookCollectionTable.repaint()
        if self.hookCollectionManager.getHookCollection(hookCollectionName) == None:
            self.hookCollectionManager.addHookCollection(HookCollection(
                hookCollectionName, executionSequence, status, description))

    def retranslateUi(self, hookCollectionDialog):
        _translate = QtCore.QCoreApplication.translate
        hookCollectionDialog.setWindowTitle(_translate("hookCollectionDialog", "Create/Edit Hook Collection"))
        self.label.setText(_translate("hookCollectionDialog", "Hook Collection Name"))
        self.label_2.setText(_translate("hookCollectionDialog", "Description"))
        self.label_3.setText(_translate("hookCollectionDialog", "Status"))
        self.label_4.setText(_translate("hookCollectionDialog", "Execution Sequence"))
        self.label_7.setText(_translate("hookCollectionDialog", "Hook Execuction Sequence"))
        self.label_6.setText(_translate("hookCollectionDialog", "Status"))
        self.label_5.setText(_translate("hookCollectionDialog", "Hook"))
        self.saveButton.setText(_translate("hookCollectionDialog", "Save"))
        self.cancelButton.setText(_translate("hookCollectionDialog", "Cancel"))